mrmustard/lab_dev/circuit_components.py

```diff
@@ -368,45 +387,81 @@ def on(self, modes: Sequence[int]) -> CircuitComponent:
+        Returns an array representation of this component in the Fock basis with the given shape.
+        If the shape is not given, it defaults to the ``auto_shape`` of the component if it is
+        available, otherwise it defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
 
-        Uses the :meth:`mrmustard.physics.converters.to_fock` method to convert the internal
-        representation.
+        Args:
+            shape: The shape of the returned representation. If ``shape`` is given as an ``int``,
+                it is broadcasted to all the dimensions. If not given, it is estimated.
+            batched: Whether the returned representation is batched or not. If ``False`` (default)
+                it will squeeze the batch dimension if it is 1.
+        Returns:
+            array: The Fock representation of this component.
+        """
+        if isinstance(shape, int):
+            shape = (shape,) * self.representation.ansatz.num_vars
+        auto_shape = self.auto_shape()
+        shape = shape or auto_shape
+        if len(shape) != len(auto_shape):
+            raise ValueError(
+                f"Expected Fock shape of length {len(auto_shape)}, got length {len(shape)}"
+            )
+
+        try:
+            As, bs, cs = self.bargmann
+            arrays = [math.hermite_renormalized(A, b, c, shape) for A, b, c in zip(As, bs, cs)]
+        except AttributeError:
+            arrays = self.representation.reduce(shape).array
+        if not batched and len(arrays) == 1:
+            arrays = arrays[0]
+        return arrays
+
+    def to_fock(self, shape=None):
+        r"""
+        Returns a new circuit component with the same attributes as this and a ``Fock`` representation.
 
         .. code-block::
 
-            >>> from mrmustard.physics.converters import to_fock
             >>> from mrmustard.lab_dev import Dgate
+            >>> from mrmustard.physics.representations import Fock
 
             >>> d = Dgate([1], x=0.1, y=0.1)
             >>> d_fock = d.to_fock(shape=3)
 
             >>> assert d_fock.name == d.name
             >>> assert d_fock.wires == d.wires
-            >>> assert d_fock.representation == to_fock(d.representation, shape=3)
+            >>> assert isinstance(d_fock.representation, Fock)
 
         Args:
             shape: The shape of the returned representation. If ``shape``is given as
                 an ``int``, it is broadcasted to all the dimensions. If ``None``, it
-                defaults to the value of ``AUTOCUTOFF_MAX_CUTOFF`` in the settings.
+                defaults to the value of ``AUTOSHAPE_MAX`` in the settings.
         """
-        return self.__class__._from_attributes(
-            to_fock(self.representation, shape=shape),
-            self.wires,
-            self.name,
-        )
+        fock = Fock(math.astensor(self.fock(shape, batched=True)), batched=True)
```

Comment:
Would it be clearer to rename `self.fock` to `self.fock_array`?

Reply:
you're right that it might be clearer, but for the other representations we use only the name (bargmann, quadrature etc)